p2p/src/net/mock/transport/channel.rs

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use once_cell::sync::Lazy;
+use tokio::sync::mpsc;
+
+use crate::net::mock::{types::Message, SocketService, TransportService};
+
+static NETWORK_HANDLE: Lazy<mpsc::UnboundedSender<SocketCommand>> = Lazy::new(|| {
+    let (tx, mut rx) = mpsc::unbounded_channel();
+    tokio::spawn(async move {
+        loop {
+            match rx.recv().await {
+                Some(cmd) => match cmd {
+                    SocketCommand::Connect(address) => {
+                        // TODO: check if peer with `address` has connected
+                    }
+                    SocketCommand::Bind(address) => {
+                        // TODO: check if `address` is free and if it is, bind this peer to that address
+                    }
+                },
+                None => {
+                    panic!("TODO")
+                }
+            }
+        }
+    });
+    tx
+});
+
+enum SocketCommand {
+    Connect(u64),
+    Bind(u64),
```

Comment:
Just a reminder but when you're implementing this functionality and you want to get the return value for, e.g., the `Connect` command, remember to add `oneshot::sender` to these enum values so the `MockNetwork` can send the return value for the peer that tried to connect.